runtime/autoload/phpcomplete.vim

```diff
@@ -2082,26 +2078,19 @@ function! phpcomplete#GetClassContentsStructure(file_path, file_lines, class_nam
 	"	...
 	" ]
 	"
-	let full_file_path = fnamemodify(a:file_path, ':p')
 	let class_name_pattern = '[a-zA-Z_\x7f-\xff\\][a-zA-Z_0-9\x7f-\xff\\]*'
-	let cfile = join(a:file_lines, "\n")
+	let full_file_path = fnamemodify(a:file_path, ':p')
 	let result = []
-	" We use new buffer and (later) normal! because
-	" this is the most efficient way. The other way
-	" is to go through the looong string looking for
-	" matching {}
-
-	" remember the window we started at
-	let phpcomplete_original_window = winnr()
-
-	silent! below 1new
-	silent! 0put =cfile
-	call search('\c\(class\|interface\|trait\)\_s\+'.a:class_name.'\(\>\|$\)')
-	let cfline = line('.')
-	call search('{')
-	let endline = line('.')
-
-	let content = join(getline(cfline, endline), "\n")
+	let popup_id = popup_create(a:file_lines, {'hidden': v:true})
+    call win_execute(popup_id, 'syntax on')
```

Comment:
Executing `:syntax on` on every completion looks inefficient, and might have undesirable effects.  And I don't think it needs to be executed from `win_execute()` since its effect should not depend from the current window.  Also, if the syntax was not enabled, it should be disabled at the end.  So, instead, you could write this (right from `phpcomplete#GetClassContentsStructure()`; no need to put this inside `win_execute()`):

    let syntax_was_enabled = exists('g:syntax_on')
    if !syntax_was_enabled
        syntax enable
    endif

    " do the stuff which requires the syntax to be enabled

    if !syntax_was_enabled
        syntax off
    endif

---

And if you need some syntax highlighting to be applied in your popup, for example `php`, then you can execute this:

    call win_execute(popup_id, 'do Syntax php')

Or maybe this:

    call win_execute(popup_id, 'set syntax=php')